sys/net/gnrc/network_layer/ndp2/gnrc_ndp2.c

```diff
@@ -0,0 +1,567 @@
+                gnrc_pktbuf_release(pkt);
+                return;
+            }
+        }
+    }
+#ifdef MODULE_GNRC_SIXLOWPAN_ND
+    /* add ARO based on interface */
+    if (netif->flags & GNRC_IPV6_NETIF_FLAGS_SIXLOWPAN) {
+        if (l2src_len != sizeof(eui64_t)) {
+            l2src_len = (uint16_t)gnrc_netapi_get(netif->pid,
+                                                  NETOPT_ADDRESS_LONG, 0,
+                                                  l2src, sizeof(l2src));
+            if (l2src_len != sizeof(eui64_t)) {
+                DEBUG("ndp2: can't get EUI-64 of the interface\n");
+                gnrc_pktbuf_release(pkt);
+                return;
+            }
+        }
+        hdr = gnrc_sixlowpan_nd_opt_ar_build(0, GNRC_SIXLOWPAN_ND_AR_LTIME,
+                                             (eui64_t *)l2src, pkt);
+        if (hdr == NULL) {
+            DEBUG("ndp2: error allocating ARO.\n");
+            gnrc_pktbuf_release(pkt);
+            return;
+        }
+        pkt = hdr;
+    }
+#endif  /* MODULE_GNRC_SIXLOWPAN_ND */
+    /* add neighbor solicitation header */
+    hdr = gnrc_ndp2_nbr_sol_build(tgt, pkt);
+    if (hdr == NULL) {
+        DEBUG("ndp2: error allocating neighbor solicitation.\n");
+        gnrc_pktbuf_release(pkt);
+        return;
+    }
+    pkt = hdr;
+    /* add remaining headers */
+    hdr = _build_headers(netif, src, dst, pkt);
+    if (hdr == NULL) {
+        DEBUG("ndp2: error adding lower-layer headers.\n");
+        gnrc_pktbuf_release(pkt);
+    }
+    else if (gnrc_netapi_dispatch_send(GNRC_NETTYPE_NDP2,
+                                       GNRC_NETREG_DEMUX_CTX_ALL, hdr) == 0) {
+        DEBUG("ndp2: unable to send neighbor solicitation\n");
+        gnrc_pktbuf_release(hdr);
+    }
+}
+
+void gnrc_ndp2_nbr_adv_send(const ipv6_addr_t *tgt, gnrc_ipv6_netif_t *netif,
+                            const ipv6_addr_t *dst, bool supply_tl2a,
+                            gnrc_pktsnip_t *ext_opts)
+{
+    ipv6_addr_t real_dst;
+    gnrc_pktsnip_t *hdr, *pkt = ext_opts;
+    uint8_t adv_flags = 0;
+
+    assert((tgt != NULL) && !ipv6_addr_is_multicast(tgt));
+    assert((netif != NULL) && (dst != NULL));
+    DEBUG("ndp2: send neighbor advertisement (iface: %" PRIkernel_pid
+          ", tgt: %s, ", netif->pid,
+          ipv6_addr_to_str(addr_str, tgt, sizeof(addr_str)));
+    DEBUG("dst: %s, supply_tl2a: %d)\n",
+          ipv6_addr_to_str(addr_str, dst, sizeof(addr_str)), supply_tl2a);
+    if ((netif->flags & GNRC_IPV6_NETIF_FLAGS_ROUTER) &&
+        (netif->flags & GNRC_IPV6_NETIF_FLAGS_RTR_ADV)) {
+        adv_flags |= NDP_NBR_ADV_FLAGS_R;
+    }
+    if (ipv6_addr_is_unspecified(dst)) {
+        memcpy(&real_dst, &ipv6_addr_all_nodes_link_local, sizeof(ipv6_addr_t));
+        supply_tl2a = true;
+    }
+    else {
+        memcpy(&real_dst, dst, sizeof(real_dst));
+        adv_flags |= NDP_NBR_ADV_FLAGS_S;
+    }
+    /* add SL2AO based on target address */
```

Comment:
/SL2AO/SLLAO/

Reply:
I don't know what you mean by *alternative spellings*. `SLLAO` is the official abbreviation for `Source Link-Layer Address Option` (cf. Appendix F of [RFC4861](https://tools.ietf.org/html/rfc4861)):
>   o Clarified router behavior when receiving a Router Solicitation
     without Source Link-Layer Address Option (SLLAO).

Reply:
> I don't know what you mean by alternative spellings

Source Link-Layer Address option / source link-layer address option instead of SLLAO.

The problem that I have with LL (which is why I tried to avoid it in GNRC when talking about the link-layer) is that it also is used for link-local, which is something completely different. So to avoid confusion my code always refers to the link-layer (aka layer 2) as L2, while LL (if used at all) always refers to link-local.

Reply:
Can we please close this discussion? IMHO this is a minor detail regarding to personal taste and the PRs building on this would require a lot of (TBH) annoying rework just to satisfy your position.

Reply:
It's not a matter of taste, but a matter of using the correct terms. Nobody outside RIOT will ever know what an `SL2O` is, but chances are high that they heard of `SLLAO`. Nevertheless, I see that you would need a lot of changes in the code and I am personally quite disappointed by moving on with `SL2O`, but let's get this merged now..

Reply:
> Nobody outside RIOT will ever know what an `SL2O` is

Me neither, I only know what a SL2AO is :P. And I'm disappointed that people outside of RIOT don't read [my wonderful doc](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/net/ndp.h#L86) to educate themselves...